File: Film.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package Entities;

import javafx.scene.image.ImageView;
import javafx.scene.media.MediaPlayer;

/**
 *
 * @author Med Aziz
 */
public class Film {
    
    int id,idadmin,idpromo;
    String nom,categorie,duree,photos,synopsis,actor,trailer;
    ImageView photo;
    MediaPlayer trailers;
    float prix;

    public Film() {
    }
    public static int Pdp;

    public static int getPdp() {
        return Pdp;
    }

    public static void setPdp(int Pdp) {
        Film.Pdp = Pdp;
    }

 public Film(int id,  String nom, String categorie, String duree, String photos, float prix,String synopsis, String actor) {
        this.id = id;
        this.nom = nom;
        this.categorie = categorie;
        this.duree = duree;
        this.photos = photos;
        this.prix = prix;
        this.synopsis=synopsis;
        this.actor=actor;
        }

    public Film(String photos) {
        this.photos=photos;
     }

    public Film(int id,  String nom, String categorie, String duree, String photos, float prix,String synopsis, String actor,String trailer) {
            this.id = id;
        this.nom = nom;
        this.categorie = categorie;
        this.duree = duree;
        this.photos = photos;
        this.prix = prix;
        this.synopsis=synopsis;
        this.actor=actor;
        this.trailer=trailer;
     }

    

    public String getTrailer() {
        return trailer;
    }

    public void setTrailer(String trailer) {
        this.trailer = trailer;
    }

    public MediaPlayer getTrailers() {
        return trailers;
    }

    public void setTrailers(MediaPlayer trailers) {
        this.trailers = trailers;
    }
    
    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public int getIdadmin() {
        return idadmin;
    }

    public void setIdadmin(int idadmin) {
        this.idadmin = idadmin;
    }

    public int getIdpromo() {
        return idpromo;
    }

    public void setIdpromo(int idpromo) {
        this.idpromo = idpromo;
    }

    public String getNom() {
        return nom;
    }

    public String getPhotos() {
        return photos;
    }

    public void setPhotos(String photos) {
        this.photos = photos;
    }

    public void setNom(String nom) {
        this.nom = nom;
    }

    public String getCategorie() {
        return categorie;
    }

    public void setCategorie(String categorie) {
        this.categorie = categorie;
    }

    public String getDuree() {
        return duree;
    }

    public void setDuree(String duree) {
        this.duree = duree;
    }

    public ImageView getPhoto() {
        return photo;
    }

    public void setPhoto(ImageView photo) {
        this.photo = photo;
    }

    

    public String getSynopsis() {
        return synopsis;
    }

    public void setSynopsis(String synopsis) {
        this.synopsis = synopsis;
    }

    public String getActor() {
        return actor;
    }

    public void setActor(String actor) {
        this.actor = actor;
    }

    public float getPrix() {
        return prix;
    }

    public void setPrix(float prix) {
        this.prix = prix;
    }

    @Override
    public String toString() {
        return "Film{" + "id=" + id + ", idadmin=" + idadmin + ", idpromo=" + idpromo + ", nom=" + nom + ", categorie=" + categorie + ", duree=" + duree + ", synopsis=" + synopsis + ", actor=" + actor + ", photo=" + photo + ", trailers=" + trailers + ", prix=" + prix + '}';
    }

   

   
    

  


   
    
    
}

File: Film.php
<?php

namespace App\Entity;
use  Symfony\Component\Validator\Constraints as Assert;
use App\Repository\FilmRepository;
use Doctrine\ORM\Mapping as ORM;

/**
 * @ORM\Entity(repositoryClass=FilmRepository::class)
 */
class Film
{
    /**
     * @ORM\Id
     * @ORM\GeneratedValue
     * @ORM\Column(type="integer")
     */
    private $id;

    /**
     * @ORM\Column(type="string", length=255)
     * @Assert\NotBlank(message="Ce champs est obligatoire")
     */
    private $nom;

    /**
     * @ORM\Column(type="string", length=255)
     * @Assert\NotBlank(message="Ce champs est obligatoire")
     */
    private $categorie;

    /**
     * @ORM\Column(type="string", length=255)
     * @Assert\NotBlank(message="Ce champs est obligatoire")
     */
    private $duree;

    /**
     * @ORM\Column(type="string", length=255)
     * @Assert\NotBlank(message="Ce champs est obligatoire")
     */
    private $prix;

    /**
     * @ORM\Column(type="string", length=255)
     * @Assert\NotBlank(message="Ce champs est obligatoire")
     */
    private $synopsis;

    /**
     * @ORM\Column(type="string", length=255)
     * @Assert\File(mimeTypes={"image/jpeg"})
     */
    private $image;

    /**
     * @ORM\ManyToOne(targetEntity=Acteur::class)
     * @ORM\JoinColumn(nullable=false)
     */
    private $actor;

    public function getId(): ?int
    {
        return $this->id;
    }

    public function getNom(): ?string
    {
        return $this->nom;
    }

    public function setNom(string $nom): self
    {
        $this->nom = $nom;

        return $this;
    }

    public function getCategorie(): ?string
    {
        return $this->categorie;
    }

    public function setCategorie(string $categorie): self
    {
        $this->categorie = $categorie;

        return $this;
    }

    public function getDuree(): ?string
    {
        return $this->duree;
    }

    public function setDuree(string $duree): self
    {
        $this->duree = $duree;

        return $this;
    }

    public function getPrix(): ?string
    {
        return $this->prix;
    }

    public function setPrix(string $prix): self
    {
        $this->prix = $prix;

        return $this;
    }

    public function getSynopsis(): ?string
    {
        return $this->synopsis;
    }

    public function setSynopsis(string $synopsis): self
    {
        $this->synopsis = $synopsis;

        return $this;
    }

    public function getImage()
    {
        return $this->image;
    }

    public function setImage($image): self
    {
        $this->image = $image;

        return $this;
    }

    public function getActor(): ?Acteur
    {
        return $this->actor;
    }

    public function setActor(?Acteur $actor): self
    {
        $this->actor = $actor;

        return $this;
    }
}
